pedsnetdcc/prepdb.py

```diff
@@ -0,0 +1,212 @@
+grant all    on               schema {{.Site}}_achilles to achilles_user;
+grant all    on               schema {{.Site}}_id_maps  to loading_user;
+alter default privileges for role loading_user in schema {{.Site}}_pedsnet grant select on tables to harvest_user, achilles_user, dqa_user, pcor_et_user, peds_staff;
+alter default privileges for role loading_user in schema {{.Site}}_pcornet grant select on tables to peds_staff;
+alter default privileges for role loading_user in schema {{.Site}}_id_maps grant select on tables to peds_staff;
+"""
+
+
+def _site_sql(site):
+    """Return a list of statements to create the schemas for a given site.
+
+    :param site: site name, e.g. 'dcc' or 'stlouis'
+    :type: str
+    :return: SQL statements
+    :rtype: list(str)
+    :raises: ValueError
+    """
+    tmpl = _sql_site_template
+    sql = tmpl.replace('{{.Site}}', site)
+    return [x for x in sql.split("\n") if x]
+
+
+_sql_other = """
+create schema if not exists vocabulary authorization dcc_owner;
+create schema if not exists dcc_ids authorization dcc_owner;
+grant all                  on schema dcc_ids    to loading_user;
+grant all                  on schema vocabulary to loading_user;
+grant usage                on schema vocabulary to achilles_user, dqa_user, pcor_et_user, harvest_user, peds_staff;
+grant select on all tables in schema vocabulary to achilles_user, dqa_user, pcor_et_user, harvest_user, peds_staff;
+alter default privileges for role loading_user in schema vocabulary grant select on tables to achilles_user, dqa_user, pcor_et_user, harvest_user, peds_staff;
+"""
+
+
+def _other_sql():
+    """Return a list of statements to create non-site schemas.
+
+    :rtype: list(str)
+    :raises: ValueError
+    """
+    sql = _sql_other
+    return [x for x in sql.split("\n") if x]
+
+
+def prepare_database(model_version, conn_str, update=False, dcc_only=False):
+    """Create a new database containing vocabulary and site schemas.
+
+    The initial conn_str is used for issuing a CREATE DATABASE statement.
+    statement.
+
+    A subsequent connection to the newly created database is established
+    in order to create schemas.
+
+    :param model_version: PEDSnet model version, e.g. X.Y.Z
+    :type: str
+    :param conn_str: libpq connection string
+    :type: str
+    :param update: assume the database is already created
+    :type: bool
+    :param dcc_only: only create schemas for `dcc` (no sites)
+    :return: True on success, False otherwise
+    """
+    logger.info({'msg': 'Starting prepdb for {0}.'.format(model_version)})
+    err = False
+    database_name = _make_database_name(model_version)
+
+    stmts = StatementList()
+
+    if not update:
+        stmts.extend(
+            [Statement(x) for x in _create_database_sql(database_name)])
+
+    stmts.extend([Statement(x) for x in _database_privileges_sql(database_name)])
+
+    stmts.serial_execute(conn_str)
+
+    for stmt in stmts:
+        if stmt.err:
```

Comment:
My take was that the error had already been logged, so it was enough to return under control with an error status.